File: src/boj/boj_2309.java
package boj;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class boj_2309 {
    static boolean find;

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int[] nums = new int[9];
        for (int i = 0; i < 9; i++) {
            nums[i] = Integer.parseInt(br.readLine());
        }

        permutation(nums, new ArrayList<>(), 0, new boolean[9]);
        br.close();
    }

    static void permutation(int[] nums, List<Integer> list, int sum, boolean[] selected) throws Exception {
        if (find) {
            return;
        } else if (list.size() == 7) {
            if (sum == 100) {
                find = true;
                printAnswer(list);
            }
            return;
        }

        for (int i = 0; i < nums.length; i++) {
            if (selected[i]) {
                continue;
            }

            selected[i] = true;
            list.add(nums[i]);
            permutation(nums, list, sum + nums[i], selected);
            list.remove(list.size() - 1);
            selected[i] = false;
        }
    }

    static void printAnswer(List<Integer> list) throws Exception {
        Collections.sort(list);

        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        for (int i : list) {
            bw.append(String.valueOf(i));
            bw.newLine();
        }
        bw.flush();
        bw.close();
    }
}

File: src/boj/boj_2309.kt
package boj

import java.io.BufferedReader
import java.io.InputStreamReader

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    var sum = 0
    val input = IntArray(9) {
        readLine().toInt().also {
            sum += it
        }
    }.sorted()

    for (i in 0 until input.size - 1) {
        for (j in i + 1 until input.size) {
            if (sum - input[i] - input[j] == 100) {
                for (k in input.indices) {
                    if (k != i && k != j) {
                        println(input[k])
                    }
                }
                return@with
            }
        }
    }
}